code/mission/missionmessage.cpp

```diff
@@ -486,45 +520,31 @@ void message_parse(MessageFormat format) {
 		}
 	}
 
+	bool require_exact_mood_match;
 	if (optional_string("$Mood:")) {
-		SCP_string buf; 
-		bool found = false;
-
-		stuff_string(buf, F_NAME); 
-		for (SCP_vector<SCP_string>::iterator iter = Builtin_moods.begin(); iter != Builtin_moods.end(); ++iter) {
-			if (*iter == buf) {
-				msg.mood = (int)std::distance(Builtin_moods.begin(), iter);
-				found = true;
-				break;
-			}
-		}
-
-		if (!found) {
-			// found a mood, but it's not in the list of moods at the start of the table
-			Warning(LOCATION, "Message.tbl has an entry for mood type %s, but this mood is not in the #Moods section of the table.", buf.c_str()); 
-		}
-	}
-	else {
-		msg.mood = 0;
+		SCP_string buf;
+		stuff_string(buf, F_NAME);
+		msg.mood = lookup_mood(buf);
```

Comment:
If a mood isn't found, msg.mood will now be set to -1 instead of 0. Which in turn means that erroneous moods for messages caused the messages to be boosted when the mood _was_ actually 0.
While I doubt that that has been taken advantage off in an existing mod, it should probably at least get a comment here acknowledging that change, in case anyone comes looking in the future.

Reply:
Would it? Looking at the code on `master`, I'm not sure that `msg.mood` is actually initialized on a missing mood. Isn't that UB?

Reply:
True, turns out I was confusing things, and the default-initialization implicitly causing a zero-initialization is only correct for variables with a static lifetime